src/sysinfo.cc

```diff
@@ -12,59 +12,67 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#include "internal_macros.h"
+#include "benchmark/benchmark.h"
+#include "internal_macros.h"  // IWYU pragma: keep
 
 #ifdef BENCHMARK_OS_WINDOWS
 #include <shlwapi.h>
 #undef StrCat  // Don't let StrCat in string_util.h be renamed to lstrcatA
 #include <versionhelpers.h>
 #include <windows.h>
+
 #include <codecvt>
-#else
-#include <fcntl.h>
-#ifndef BENCHMARK_OS_FUCHSIA
-#include <sys/resource.h>
-#endif
-#include <sys/time.h>
-#include <sys/types.h>  // this header must be included before 'sys/sysctl.h' to avoid compilation error on FreeBSD
+#include <memory>
+#endif  // ifdef BENCHMARK_OS_WINDOWS
+
+#ifndef BENCHMARK_OS_WINDOWS
 #include <unistd.h>
+// IWYU pragma: no_include <bits/local_lim.h>
+#endif  // ifndef BENCHMARK_OS_WINDOWS
+
 #if defined BENCHMARK_OS_FREEBSD || defined BENCHMARK_OS_MACOSX || \
     defined BENCHMARK_OS_NETBSD || defined BENCHMARK_OS_OPENBSD
+#include <sys/types.h>  // this header must be included before 'sys/sysctl.h' to avoid compilation error on FreeBSD
 #define BENCHMARK_HAS_SYSCTL
 #include <sys/sysctl.h>
-#endif
-#endif
+#endif  //  defined BENCHMARK_OS_FREEBSD || defined BENCHMARK_OS_MACOSX ||
+        //  defined BENCHMARK_OS_NETBSD || defined BENCHMARK_OS_OPENBSD
+
 #if defined(BENCHMARK_OS_SOLARIS)
 #include <kstat.h>
-#endif
+#endif  //  defined(BENCHMARK_OS_SOLARIS)
+
 #if defined(BENCHMARK_OS_QNX)
 #include <sys/syspage.h>
-#endif
+#endif  // defined(BENCHMARK_OS_QNX)
 
 #include <algorithm>
 #include <array>
 #include <bitset>
-#include <cerrno>
+#include <cctype>
 #include <climits>
 #include <cstdint>
 #include <cstdio>
 #include <cstdlib>
-#include <cstring>
-#include <fstream>
+#include <cstring>  // IWYU pragma: keep
+#include <fstream>  // IWYU pragma: keep
 #include <iostream>
-#include <iterator>
-#include <limits>
+#include <string>
+#include <utility>
+#include <vector>
+
+#ifdef BENCHMARK_HAS_SYSCTL
```

Comment:
this shouldn't need to be guarded...

Reply:
The functions provided by that header are only used within `#ifdef BENCHMARK_HAS_SYSCTL` block, i believe

Reply:
i think that's too hard to know, and hard to maintain. including memory and check when they're not used isn't a significant enough cost to harm maintainability of the code.

Reply:
Can you be more specific please, are you worried about NOT including the headers needed (that will result in compilation error), or including unneeded headers?

Reply:
i'm worried about the first bit: that we end up not including <memory> or "check" and they end up being needed either by the existing code or by a future change. it's easier to just include them.

Reply:
Well, and if future change requires boost includes so what then, add them now?
If include is needed it will need to be added, or else the code won't compile, it's impossible to miss :)

Reply:
that's a fallacy. memory and check _are_ needed today for some configurations. it's easier to add them now globally than to remember to add them later (and it's not impossible to miss, depending on configuration and platform).

Reply:
> memory and check _are_ needed today for some configurations.

I fully agree, the key word being *some*.

> it's easier to add them now globally than to remember to add them later

> (and it's not impossible to miss, depending on configuration and platform).

I believe i'm missing some reasoning here. How can that happen?
The functionality of any C++ STL header is mandated by the standard,
If it provides too much, then we won't have any issues,
if it doesn't provide what it should then it's simply an non-compliant STL implementation.

Reply:
if we don't cover a configuration in travis/appveyor then we rely on the developer checking their build. it wouldn't be great if they didn't, but we can't be sure they did.

i'd just rather not conditionally include c++ standard headers.